rts/Sim/Units/UnitTypes/Builder.cpp

```diff
@@ -722,15 +722,19 @@ bool CBuilder::StartBuild(BuildInfo& buildInfo, CFeature*& feature, bool& inWait
 			// closest to us
 			const CGroundBlockingObjectMap::BlockingMapCell& cell = groundBlockingObjectMap.GetCellUnsafeConst(buildInfo.pos);
 
-			const CSolidObject* o = nullptr;
 			const CUnit* u = nullptr;
 
 			// look for any blocking assistable buildee at build.pos
 			for (size_t i = 0, n = cell.size(); i < n; i++) {
-				const CUnit* cu = dynamic_cast<const CUnit*>(o = cell[i]);
+				const CUnit* cu = dynamic_cast<const CUnit*>(cell[i]);
 
 				if (cu == nullptr)
 					continue;
+				/* NB: allies (cf. `teamHandler.Ally`) are also blocked
+				 * even though in theory we could want to assist them. This
+				 * is because allies aren't caught by `GetClosestFriendlyUnit`
+				 * so it would be possible to build on top of them if they
+				 * weren't assistable. */
```

Comment:
Can you further elaborate on this comment?

I've looked in the code a bit and it seems that this invokes
https://github.com/spring/spring/blob/develop/rts/Game/GameHelper.cpp#L801-L806
which in turn uses the Friendly class to filter units. The per-team filter there uses the function you mentioned https://github.com/spring/spring/blob/develop/rts/Game/GameHelper.cpp#L406 , so I'm not sure how they "aren't getting caught"?

Reply:
Now I'm unsure. Maybe there was some issue with the tests. I think it would be the best if I just remove the comment and maybe come up with a better one later if the situation clears up.